"""Category name should not be unique by itself

Revision ID: f69e0dcbeb12
Revises: ff117ff70158
Create Date: 2024-09-15 20:42:09.398151

"""
from alembic import op
import sqlalchemy as sa


# revision identifiers, used by Alembic.
revision = 'f69e0dcbeb12'
down_revision = 'ff117ff70158'
branch_labels = None
depends_on = None


def upgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('category', schema=None) as batch_op:
        batch_op.drop_constraint('category_name_key', type_='unique')
        batch_op.create_unique_constraint(None, ['user_id', 'name'])

    # ### end Alembic commands ###


def downgrade():
    # ### commands auto generated by Alembic - please adjust! ###
    with op.batch_alter_table('category', schema=None) as batch_op:
        batch_op.drop_constraint(None, type_='unique')
        batch_op.create_unique_constraint('category_name_key', ['name'])

    # ### end Alembic commands ###
